### PR TITLE
New param Max

### PR DESCRIPTION
**Pull request by Maykonn**
Max param will set the maximum quantity of chosen attributes must to be filled.
If you have 3 attributes but just 1 must to be filled then set `max=1`. Default value for max is 0(all attributes can be filled).

### CODE REVIEW

**Comment by Maykonn** [comment]
Resolving #11

**Comment by Maykonn** [comment]
Hello!

Someone can merge this pull request on master?

**Comment by sdlins** [comment]
Sorry, I have been very busy studying other things last months... I will review your PR and merge if it is ok. Thank you so much for your contribuition.

File: AtLeastValidator.php [hunk 1]
```diff
@@ -55,6 +55,12 @@ class AtLeastValidator extends Validator
     public $min = 1;
 
     /**
+     * @var integer the maximun quantity of chosen attributes that must to be filled.
+     * Defaults to 1.
```
Comment (sdlins) [inline AtLeastValidator.php]: A minor error here: Defaults to 0;

File: AtLeastValidator.php [hunk 2]
```diff
@@ -99,10 +105,14 @@ public function validateAttribute($model, $attribute)
             $chosen += !empty($value) ? 1 : 0;
         }
 
-        if (!$chosen || $chosen < $this->min) {
+        if (
+            !$chosen || $chosen < $this->min ||
+            ($this->max > 0 and $chosen > $this->max)
```
Comment (sdlins) [inline AtLeastValidator.php]: please, change `and` to `&&`

**Review by sdlins — COMMENTED** [review]
Please, remove the composer.lock off the set and re-send the PR after the other minor things commented on the code. Thank you so much for your contribuition. A propósito, sou do PR tmb. T+

File: AtLeastValidator.php [hunk 3]
```diff
@@ -107,7 +107,7 @@ public function validateAttribute($model, $attribute)
 
         if (
             !$chosen || $chosen < $this->min ||
-            ($this->max > 0 and $chosen > $this->max)
```
Comment (sdlins) [inline AtLeastValidator.php]: thnx

File: AtLeastValidator.php [hunk 4]
```diff
@@ -88,6 +88,9 @@ public function init()
         }
         if ($this->message === null) {
             $this->message = 'You must fill at least {min} of the attributes {attributes}.';
+            if($this->max > 0) {
```
Comment (sdlins) [inline AtLeastValidator.php]: Could you remove the dot from the end of the L90 and make a more simple sintax, please?
Suggestion:
`$this->message .= $this->max > 0 ? ' and at most {max} attribute(s).' : '.';`
Or you could even think a better way to generate a better error message for cases when `$min=0` and `$max=2`, for example.

Thank you so much.

**Comment by sdlins** [comment]
@Maykonn, thank you for your work. I will wait you fix the minor details appointed here and resend the PR.

**Review by sdlins — CHANGES_REQUESTED** [review]
Please, check the feedback before I can merge your changes. Thank you!

**Comment by Maykonn** [comment]
@slinstj Done :smile:

**Comment by sdlins** [comment]
It will be closed till it is ok. Thank you for your contribution!

**Comment by Maykonn** [comment]
@slinstj The pull-request was accepted?

**Comment by sdlins** [comment]
@Maykonn, it was not accepted for now. Please, read the messages I attached into the code. Thank you.

**Comment by Maykonn** [comment]
I don't understand. See https://github.com/slinstj/yii2-at-least-validator/pull/12/commits/b418475e99cd5ffbf35745e3020a2904523369f6

**Comment by sdlins** [comment]
I have seen. Please, see my comments [*in your PR*](https://github.com/slinstj/yii2-at-least-validator/pull/12/files/b418475e99cd5ffbf35745e3020a2904523369f6). Thank you again dude!

**Pull request closed without merging**